Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -724,6 +736,13 @@ namespace Terrain
                 const uint16_t coord = y * request.m_samplesX + x;
 
                 const float height = heights.at(queryCoord);
+                if (height < 0.0)
+                {
+                    // Terrain doesn't exist at this point
+                    meshHeights.at(coord) = 0xFFFF;
```

Comment:
Nit: Same here, should probably have a named constant. For this one, it also might be worth an extra comment explaining that because we only use 15 bits of height data, this value can't be legitimately generated as a height.